Uni_life_tracker/app/src/main/java/com/generic/plannr/Gateways/UserGateway.java

```diff
@@ -0,0 +1,267 @@
+package com.generic.plannr.Gateways;
+
+import android.annotation.SuppressLint;
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import com.generic.plannr.Database.DatabaseClient;
+import com.generic.plannr.Entities.Event;
+import com.generic.plannr.Entities.User;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserGateway implements UserGatewayInterface {
+
+    public SQLiteDatabase db;
+    public DatabaseClient dbclient;
+
+    public UserGateway(Context context) {
+        dbclient = new DatabaseClient(context);
+    }
+
+    /**
+     * Open the database for reading or writing
+     */
+    public void openDatabase() {
+        db = dbclient.getWritableDatabase();
+    }
+
+    /**
+     * Insert user's info into the database
+     *
+     * @param user The User to be inserted
+     */
+    public void saveToDatabase(final User user) {
+        openDatabase();
+        ContentValues cv = new ContentValues();
+        cv.put("NAME", user.getName());
+        cv.put("UNIVERSITY", user.getSchool());
+        cv.put("EMAIL", user.getEmail());
+        cv.put("PASSWORD", user.getPassword());
+        cv.put("LOGGEDIN", 1);
+        db.insert("userinfo", null, cv);
+    }
+
+    /**
+     * Update the User's name in the database
+     *
+     * @param name The user's name to be inserted
+     */
+    @SuppressLint("Recycle")
+    public void updateName(String name){
+        openDatabase();
+        ContentValues cv = new ContentValues();
+        cv.put("NAME", name);
+        db.update("userinfo", cv, "LOGGEDIN = 1", null);
+    }
+
+    /**
+     * Update the User's university in the database
+     *
+     * @param uni The user's university to be inserted
+     */
+    public void updateUni(String uni){
+        openDatabase();
+        ContentValues cv = new ContentValues();
+        cv.put("UNIVERSITY", uni);
+        db.update("userinfo", cv, "LOGGEDIN = 1", null);
+    }
+
+    /**
+     * Get the User associated with id userID currently stored
+     * in the database
+     *
+     * @param userID the id of the User we want to return
+     *
+     * @return the User with id userID
+     */
+    public User getByID(final int userID) {
+        openDatabase();
+        @SuppressLint("Recycle") Cursor cur = db.rawQuery("SELECT * FROM userinfo WHERE " +
+                        "ID = " + userID,
+                null);
+
+        if (!cur.moveToFirst()) {
```

Comment:
Maybe u could put this if statement into a method since it is being used multiple times

Reply:
I don't think it would reduce the amount of code since it's just one line, but I can add a comment to explain what it does